In [1]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os
import plotly.express as px

In [2]:
df = pd.read_csv(os.path.join("..","data/raw", "FAO.csv"), encoding='ISO-8859-1')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21477 entries, 0 to 21476
Data columns (total 63 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Area Abbreviation  21477 non-null  object 
 1   Area Code          21477 non-null  int64  
 2   Area               21477 non-null  object 
 3   Item Code          21477 non-null  int64  
 4   Item               21477 non-null  object 
 5   Element Code       21477 non-null  int64  
 6   Element            21477 non-null  object 
 7   Unit               21477 non-null  object 
 8   latitude           21477 non-null  float64
 9   longitude          21477 non-null  float64
 10  Y1961              17938 non-null  float64
 11  Y1962              17938 non-null  float64
 12  Y1963              17938 non-null  float64
 13  Y1964              17938 non-null  float64
 14  Y1965              17938 non-null  float64
 15  Y1966              17938 non-null  float64
 16  Y1967              179

# Découverte
Nous avons un dataset sur la production de denrées dans le monde.

On apprend donc les informations de bases sur ce dataset, compris avec **63 attributs** et **21477 observations**
On peut séparer ces attributs en 2 partie : les **attributs sur le produit** et les **attributs temporels**
On constate d'ailleurs que les valeurs manquantes sont uniquement présentes dans les attributs temporels, qui pourrait traduire simplement l'abscence de production (à explorer)


In [3]:
df.head(3)

Area Abbreviation  Area Code         Area  Item Code  \
0               AFG          2  Afghanistan       2511   
1               AFG          2  Afghanistan       2805   
2               AFG          2  Afghanistan       2513   

                       Item  Element Code Element         Unit  latitude  \
0        Wheat and products          5142    Food  1000 tonnes     33.94   
1  Rice (Milled Equivalent)          5142    Food  1000 tonnes     33.94   
2       Barley and products          5521    Feed  1000 tonnes     33.94   

   longitude  ...   Y2004   Y2005   Y2006   Y2007   Y2008   Y2009   Y2010  \
0      67.71  ...  3249.0  3486.0  3704.0  4164.0  4252.0  4538.0  4605.0   
1      67.71  ...   419.0   445.0   546.0   455.0   490.0   415.0   442.0   
2      67.71  ...    58.0   236.0   262.0   263.0   230.0   379.0   315.0   

    Y2011  Y2012  Y2013  
0  4711.0   4810   4895  
1   476.0    425    422  
2   203.0    367    360  

[3 rows x 63 columns]

# Organisation

Le dataset est rangé par pays, puis chaque ligne correspond à un produit. On y associe les informations sur le pays (position, abréviation,...) et les informations sur le produit (nom, type de produit) surtout avec sa production par année (Y2004, Y2005, Y....)  
Avant de combler les valeurs vides, on vérifie si on peut enlever des colonnes inutile : Unit possède vraissemblablement la même valeur (à vérifier), abréviation et le code ne nous sont pas utiles, l'élément et le produit code non plus et toute les colonnes Y... pourrait n'en former qu'une (après avoir gérer les valeurs nuls)

In [4]:
print(df["Unit"].unique())

['1000 tonnes']


In [5]:
dropped_cols = ["Area Abbreviation", "Area Code", "Item Code", "Element Code", "Unit"]
df.drop(dropped_cols, axis=1, inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21477 entries, 0 to 21476
Data columns (total 58 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Area       21477 non-null  object 
 1   Item       21477 non-null  object 
 2   Element    21477 non-null  object 
 3   latitude   21477 non-null  float64
 4   longitude  21477 non-null  float64
 5   Y1961      17938 non-null  float64
 6   Y1962      17938 non-null  float64
 7   Y1963      17938 non-null  float64
 8   Y1964      17938 non-null  float64
 9   Y1965      17938 non-null  float64
 10  Y1966      17938 non-null  float64
 11  Y1967      17938 non-null  float64
 12  Y1968      17938 non-null  float64
 13  Y1969      17938 non-null  float64
 14  Y1970      17938 non-null  float64
 15  Y1971      17938 non-null  float64
 16  Y1972      17938 non-null  float64
 17  Y1973      17938 non-null  float64
 18  Y1974      17938 non-null  float64
 19  Y1975      17938 non-null  float64
 20  Y1976 

# Valeurs nuls
Maintenant avant de créer la colonne regroupant toute les années, il faut combler les valeurs nuls.  
Pour ça plusieurs choix : la médiane, la moyenne, le mode ou une valeur par défaut (0, 1, 36, ...).  
On constate que plus l'année est vieille, plus il y a de valeur nul. Ainsi on peut savoir que les valeurs nuls correspondent au moment avant la production (ou avant qu'on trace la production). Donc on peut se demander comment combler les valeurs nuls ?  
La médiane et la moyenne ne sont pas sencé, avec une évolution ce serait bizarre de commencer par la médiane puis de redescendre...  
Le mode subit la même problématique.  
Enfin il parraît logique de combler les valeur vides avec 0, pour montrer l'absence de production.

In [6]:
df.fillna(0, inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21477 entries, 0 to 21476
Data columns (total 58 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Area       21477 non-null  object 
 1   Item       21477 non-null  object 
 2   Element    21477 non-null  object 
 3   latitude   21477 non-null  float64
 4   longitude  21477 non-null  float64
 5   Y1961      21477 non-null  float64
 6   Y1962      21477 non-null  float64
 7   Y1963      21477 non-null  float64
 8   Y1964      21477 non-null  float64
 9   Y1965      21477 non-null  float64
 10  Y1966      21477 non-null  float64
 11  Y1967      21477 non-null  float64
 12  Y1968      21477 non-null  float64
 13  Y1969      21477 non-null  float64
 14  Y1970      21477 non-null  float64
 15  Y1971      21477 non-null  float64
 16  Y1972      21477 non-null  float64
 17  Y1973      21477 non-null  float64
 18  Y1974      21477 non-null  float64
 19  Y1975      21477 non-null  float64
 20  Y1976 

Maintenant nous pouvons créer cette colonne.

In [7]:
def years_to_array(row):
    return [row["Y"+str(i)] for i in range(1961,2014)]

df["years_evolution"] = df.apply(years_to_array, axis=1)
df.head(3)

Area                      Item Element  latitude  longitude   Y1961  \
0  Afghanistan        Wheat and products    Food     33.94      67.71  1928.0   
1  Afghanistan  Rice (Milled Equivalent)    Food     33.94      67.71   183.0   
2  Afghanistan       Barley and products    Feed     33.94      67.71    76.0   

    Y1962   Y1963   Y1964   Y1965  ...   Y2005   Y2006   Y2007   Y2008  \
0  1904.0  1666.0  1950.0  2001.0  ...  3486.0  3704.0  4164.0  4252.0   
1   183.0   182.0   220.0   220.0  ...   445.0   546.0   455.0   490.0   
2    76.0    76.0    76.0    76.0  ...   236.0   262.0   263.0   230.0   

    Y2009   Y2010   Y2011  Y2012  Y2013  \
0  4538.0  4605.0  4711.0   4810   4895   
1   415.0   442.0   476.0    425    422   
2   379.0   315.0   203.0    367    360   

                                     years_evolution  
0  [1928.0, 1904.0, 1666.0, 1950.0, 2001.0, 1808....  
1  [183.0, 183.0, 182.0, 220.0, 220.0, 195.0, 231...  
2  [76.0, 76.0, 76.0, 76.0, 76.0, 75.0, 71.0, 72....  

[3 rows x 59 columns]

Enfin, nous pouvons supprimer les colonnes "Y...."

In [8]:
years_cols = ["Y"+str(i)  for i in range(1961,2014)]
df.drop(columns=years_cols, inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21477 entries, 0 to 21476
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Area             21477 non-null  object 
 1   Item             21477 non-null  object 
 2   Element          21477 non-null  object 
 3   latitude         21477 non-null  float64
 4   longitude        21477 non-null  float64
 5   years_evolution  21477 non-null  object 
dtypes: float64(2), object(4)
memory usage: 1006.9+ KB


Nous avons donc un bon dataset qui sera plus agréable à la lecture et l'utilisation

In [9]:
df.head()

Area                      Item Element  latitude  longitude  \
0  Afghanistan        Wheat and products    Food     33.94      67.71   
1  Afghanistan  Rice (Milled Equivalent)    Food     33.94      67.71   
2  Afghanistan       Barley and products    Feed     33.94      67.71   
3  Afghanistan       Barley and products    Food     33.94      67.71   
4  Afghanistan        Maize and products    Feed     33.94      67.71   

                                     years_evolution  
0  [1928.0, 1904.0, 1666.0, 1950.0, 2001.0, 1808....  
1  [183.0, 183.0, 182.0, 220.0, 220.0, 195.0, 231...  
2  [76.0, 76.0, 76.0, 76.0, 76.0, 75.0, 71.0, 72....  
3  [237.0, 237.0, 237.0, 238.0, 238.0, 237.0, 225...  
4  [210.0, 210.0, 214.0, 216.0, 216.0, 216.0, 235...

# Exploration des valeurs

In [10]:
print(df["Area"].unique(), len(df["Area"].unique()))

['Afghanistan' 'Albania' 'Algeria' 'Angola' 'Antigua and Barbuda'
 'Argentina' 'Armenia' 'Australia' 'Austria' 'Azerbaijan' 'Bahamas'
 'Bangladesh' 'Barbados' 'Belarus' 'Belgium' 'Belize' 'Benin' 'Bermuda'
 'Bolivia (Plurinational State of)' 'Bosnia and Herzegovina' 'Botswana'
 'Brazil' 'Brunei Darussalam' 'Bulgaria' 'Burkina Faso' 'Cabo Verde'
 'Cambodia' 'Cameroon' 'Canada' 'Central African Republic' 'Chad' 'Chile'
 'China, Hong Kong SAR' 'China, Macao SAR' 'China, mainland'
 'China, Taiwan Province of' 'Colombia' 'Congo' 'Costa Rica'
 "Côte d'Ivoire" 'Croatia' 'Cuba' 'Cyprus' 'Czechia'
 "Democratic People's Republic of Korea" 'Denmark' 'Djibouti' 'Dominica'
 'Dominican Republic' 'Ecuador' 'Egypt' 'El Salvador' 'Estonia' 'Ethiopia'
 'Fiji' 'Finland' 'France' 'French Polynesia' 'Gabon' 'Gambia' 'Georgia'
 'Germany' 'Ghana' 'Greece' 'Grenada' 'Guatemala' 'Guinea' 'Guinea-Bissau'
 'Guyana' 'Haiti' 'Honduras' 'Hungary' 'Iceland' 'India' 'Indonesia'
 'Iran (Islamic Republic of)' 'Iraq' 'I

In [11]:
print(df["Element"].unique())

['Food' 'Feed']


In [12]:
print(len(df["Item"].unique()), len(df[df["Element"]=="Food"]["Item"]) * 100 /len(df))

115 81.6128882059878


In [35]:
def get_year_production(liste: list, index: int) :
    return liste[index]

print(df["years_evolution"].apply(get_year_production, args=(0,)).sum(), df["years_evolution"].apply(get_year_production, args=(52,)).sum())

3502611.0 12361248


# Détails Datasets

Nous avons **21477** observations, chacune qui caractérise la production d'un produit par un pays pendant 53 ans.  
Nous avons 174 pays présents dans le dataset, accompagné de leurs positions géographique.  
Il y a 115 produits, dont environ 81,6% sont de la nourritures, le reste étant pour nourrir le bétail.  
On constate une augmentation de la production grlobale en 52 ans, passant d'environ **3 502 611 000** T produite en 1961, avec **12 361 248 000** T en 2013 (presque x4)